tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/NewFunctionsSuite.scala

```diff
@@ -637,6 +637,62 @@ class NewFunctionsSuite extends BaseDottySuite {
     )
   }
 
+  test("dependent-type-arrow-after-nl") {
+    val expectedSyntax = "type T = (e: Entry) => e.Key"
+    val expectedTree = Defn.Type(
+      Nil,
+      pname("T"),
+      Nil,
+      Type.Function(
+        List(Type.TypedParam(pname("e"), pname("Entry"))),
+        Type.Select(tname("e"), pname("Key"))
+      ),
+      Type.Bounds(None, None)
+    )
+
+    runTestAssert[Stat](
+      """|type T = 
+         |  (e: Entry) 
+         |  => e.Key
```

Comment:
This one is valid, the ones with lower indentation are not from what I checked.

Reply:
thank you for checking. in that case, looks like the other solution (#3021) is more appropriate here, please take a look.